src/components/modals/EditDialogModal.tsx

```diff
@@ -303,11 +331,27 @@ class EditDialogModal extends React.Component<Props, ComponentState> {
             }
             // Otherwise continue
             else {
+                // TEMP: until server can exclude label conflicts with self
+                if (this.showInternalLabelConflict()) {   
+                    return
+                }
                 await this.props.onContinueDialog(newTrainDialog, userInput)
             }
         }
     }
 
+    // TEMP: until server can exclude label conflicts with self, we need
+    // to check for them and force save before we can add a turn 
+    showInternalLabelConflict(): boolean {
```

Comment:
Minor but the naming on this is misleading: `showInternalLabelConflict` doesn't always show the conflict.  Since it's temporary I guess it won't be in the code long but that's what they always say.

The methods that consume or use this treat it like another `hasInternalLabelConflict` which is clear about returning true or false and don't care if it opens the modal or not they just return early to prevent execution

There is similar code:
```
            if (await this.hasConflicts(textVariations)) {
                return
            }
```

Not sure about new name though if you call it `hasInternalLabelConflict` to wrap the DialogUtil method it doesn't indicate it will open the modal